### ISSUE TITLE
Allow session recording enabling/disabling on a role basis

### ISSUE THREAD

**Issue by deusxanima** [gravitational/teleport#8752]
### What
Customers have requested the ability to enable/disable session recordings on a per-role basis.

Desktop Access already supports this. Add support Server Access and Kubernetes Access. Stretch goal to support Application Access and Database Access.

### Why
Customers who add Teleport to their monitoring tools may create automation/monitoring users in Teleport which access all Teleport nodes with the identity file at a certain interval to ensure that all nodes are accessible. The problem is that doing this causes backend logs and sessions to increase exponentially, even though the sessions to nodes are not interactive.

**Comment by sysadmiral** [comment]
We were talking to Sean at DevOpsDays Birmingham about this and I would just add that per-session would also be super useful like `tsh ssh --no-record` or something?

**Comment by zmb3** [comment]
@gabrielcorado do your new recording mode role options make it possible to disable session recording on a per-role basis?

**Comment by gabrielcorado** [comment]
No, we've considered disabling/enabling session recordings out of the feature scope. But we can reuse the options added to cover this.

**Comment by adamcarheden-teleport** [comment]
Requested modes for Kubernetes: `strict|best_effort|off` (currently, ssh is strict|best_effort, desktop is on|off)
